sql/core/src/test/scala/org/apache/spark/sql/DatasetSuite.scala

```diff
@@ -1586,6 +1591,30 @@ class DatasetSuite extends QueryTest with SharedSQLContext {
       }
     }
   }
+
+  test("SPARK-34726: Fix collectToPython timeouts") {
+    val listener = new QueryExecutionListener {
+      override def onFailure(funcName: String, qe: QueryExecution, exception: Exception): Unit = {}
+
+      override def onSuccess(funcName: String, qe: QueryExecution, duration: Long): Unit = {
+        // Longer than 15s in `PythonServer.setupOneConnectionServer`
+        Thread.sleep(20 * 1000)
```

Comment:
Just for my own understanding does this mean the test waits 15 seconds to pass?

Reply:
Yes, we need to wait a bit longer than the timeout. Without the fix in `Dataset.collectToPython` this UT fails.